src/maidsafe/vault/vault.h

```diff
@@ -39,18 +42,38 @@ namespace maidsafe {
 
 namespace vault {
 
+//// Helper function to parse data name and contents
+//// FIXME this need discussion, adding it temporarily to progress
+// template <typename ParsedType>
+// ParsedType ParseData(const SerialisedData& serialised_data) {
+//  InputVectorStream binary_input_stream{serialised_data};
+//  typename ParsedType::Name name;
+//  typename ParsedType::serialised_type contents;
+//  Parse(binary_input_stream, name, contents);
+//  return ParsedType(name, contents);
+// }
+
+// Helper function to parse Mpid messaging related messages : message, alert, etc.
+// FIXME this need discussion, adding it temporarily to progress
+template <typename ParsedType>
+ParsedType ParseMpidMessaging(const SerialisedData& serialised_data) {
```

Comment:
Don't think we need this helper if we use the Cereal functions.  Just call `Parse` where you call `ParseMpidMessaging` and it should work.